Find a functionally equivalent code:android.hardware.SensorManager.getRotationMatrixFromVector:COMMENT
Method Modifier: public      static      
Comment:/**
 * Helper function to convert a rotation vector to a rotation matrix.
 * Given a rotation vector (presumably from a ROTATION_VECTOR sensor), returns a
 * 9  or 16 element rotation matrix in the array R.  R must have length 9 or 16.
 * If R.length == 9, the following matrix is returned:
 * <pre>
 * /  R[ 0]   R[ 1]   R[ 2]   \
 * |  R[ 3]   R[ 4]   R[ 5]   |
 * \  R[ 6]   R[ 7]   R[ 8]   /
 * </pre>
 * If R.length == 16, the following matrix is returned:
 * <pre>
 * /  R[ 0]   R[ 1]   R[ 2]   0  \
 * |  R[ 4]   R[ 5]   R[ 6]   0  |
 * |  R[ 8]   R[ 9]   R[10]   0  |
 * \  0       0       0       1  /
 * </pre>
 * @param rotationVector the rotation vector to convert
 * @param R an array of floats in which to store the rotation matrix
 */

@@ -3,9 +3,9 @@
     float q1 = rotationVector[0];
     float q2 = rotationVector[1];
     float q3 = rotationVector[2];
-    if (rotationVector.length == 4) {
+    if (rotationVector.length >= 4) {
         q0 = rotationVector[3];
     } else {
         q0 = 1 - q1 * q1 - q2 * q2 - q3 * q3;

